client/instances.go

```diff
@@ -453,27 +457,73 @@ func (c *Client) Updates(opts *UpdatesOptions) error {
 }
 
 // Export launch the creation of a tarball to export data from an instance.
-func (c *Client) Export(domain string) error {
-	if !validDomain(domain) {
-		return fmt.Errorf("Invalid domain: %s", domain)
+func (ac *AdminClient) Export(opts *ExportOptions) error {
+	if !validDomain(opts.Domain) {
+		return fmt.Errorf("Invalid domain: %s", opts.Domain)
 	}
-	_, err := c.Req(&request.Options{
-		Method:     "POST",
-		Path:       "/instances/" + url.PathEscape(domain) + "/export",
-		NoResponse: true,
+
+	res, err := ac.Req(&request.Options{
+		Method: "POST",
+		Path:   "/instances/" + url.PathEscape(opts.Domain) + "/export",
+		Queries: url.Values{
+			"path": []string{opts.LocalPath},
+		},
 	})
-	return err
+	if err != nil {
+		return err
+	}
+
+	if opts.WaitForResult {
+		ic, err := ac.NewInstanceClient(opts.Domain, consts.Jobs)
+		if err != nil {
+			return err
+		}
+		channel, err := ic.RealtimeClient(RealtimeOptions{
+			DocTypes: []string{"io.cozy.jobs"},
+		})
+		if err != nil {
+			return err
+		}
+		defer func() {
+			channel.Close()
+		}()
+
+		defer res.Body.Close()
```

Comment:
I don't understand why we close the response body only if we wait for the result. We should do it on both cases (waiting the result, or not).

Reply:
I thought it wasn't necessary as long as we were not reading it. I moved the defer call right after getting the response (if there aren't any errors).